src/components/Questionnaire/QuestionnaireEditor.tsx

```diff
@@ -889,7 +889,15 @@ function QuestionEditor({
     value: Question[K],
     additionalFields?: Partial<Question>,
   ) => {
-    onChange({ ...question, [field]: value, ...additionalFields });
+    const updateQuestion = { ...question, [field]: value, ...additionalFields };
+    if (
+      field === "type" &&
+      value === "structured" &&
+      !updateQuestion.structured_type
+    ) {
+      updateQuestion.structured_type = "allergy_intolerance";
```

Comment:
why does this generic component have to set the stuctured_type to a very specific "allergy intolerance"? 🤔